packages/headless/src/database/mongodb/cart/cart.model.ts

```diff
@@ -0,0 +1,29 @@
+import { model, Schema } from 'mongoose';
+import { Cart } from 'src/entity/cart';
+
+const CartSchema = new Schema<Cart>(
+  {
+    id: {
+      type: String,
+      unique: true,
+    },
+    userId: String,
+    items: [
+      {
+        productId: String,
+        quantity: {
+          type: Number,
+          default: 1,
```

Comment:
remove default value. Otherwise when you will create a cart without items, then mongoose will automatically add {quantity: 1} to your array, even if you dont pass any value for items